presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1487,6 +1490,30 @@ public ConnectorTableHandle beginStatisticsCollection(ConnectorSession session,
         return tableHandle;
     }
 
+    private List<String> normalizeKey(List<String> key, List<Type> partitionTypes)
+    {
+        checkArgument(
+                key.size() == partitionTypes.size(),
+                "Key length (%d) does not match the number of columns (%d)",
+                key.size(),
+                partitionTypes.size());
+        List<String> newKey = new ArrayList<String>();
+        for (int i = 0; i < partitionTypes.size(); i++) {
+            String keyValue = key.get(i);
+            Type type = partitionTypes.get(i);
+            BlockBuilder blockBuilder = type.createBlockBuilder(null, 1, 0);
```

Comment:
I don't think we need a BlockBiulder here.

Reply:
I fixed this.